packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -682,12 +685,30 @@ function detectPnpmVersion(
   }
 }
 
-function shouldUseNpm7(
-  lockfileVersion: number | undefined,
-  nodeVersion: NodeVersion | undefined
-): boolean {
-  if (lockfileVersion === undefined) return false;
-  return lockfileVersion >= 2 && (nodeVersion?.major || 0) < 16;
+function validLockfileForPackageManager(
+  cliType: CliType,
+  lockfileVersion: number,
+  packageManagerMajorVersion: number
+) {
+  switch (cliType) {
+    case 'npm':
+    case 'bun':
+    case 'yarn':
+      return true;
+    case 'pnpm':
+      switch (packageManagerMajorVersion) {
+        case 9:
+          return [6.0, 7.0, 9.0].includes(lockfileVersion);
```

Comment:
nitpick: Remove unnecessary `.0`s

Reply:
I don't mind being explicit, but either is fine with me